src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -433,6 +433,7 @@ public enum Mode { STARTING, NORMAL, JOINING, LEAVING, DECOMMISSIONED, MOVING, D
     private static final boolean allowSimultaneousMoves = CONSISTENT_SIMULTANEOUS_MOVES_ALLOW.getBoolean();
     private static final boolean joinRing = JOIN_RING.getBoolean();
     private boolean replacing;
+    private volatile boolean hasDecommissionFailed = false;
```

Comment:
I think we can change the name to isDecommissionFailed as src/java/org/apache/cassandra/service/StorageServiceMBean.java  method isDecommissionFailed();
Edit :
can we just AtomicBoolean